analyzer/codechecker_analyzer/buildlog/log_parser.py

```diff
@@ -443,47 +443,103 @@ def get_compiler_standard(compiler, language):
 
+        c_lang = "c"
+        cpp_lang = "c++"
+
+        compiler = details['compiler']
         if compiler_info_file and os.path.exists(compiler_info_file):
             # Compiler info file exists, load it.
-            ICI.load_compiler_info(compiler_info_file, details['compiler'])
+            ICI.load_compiler_info(compiler_info_file, compiler,
+                                   c_lang, cpp_lang)
         else:
             # Invoke compiler to gather implicit compiler info.
-            if details['compiler'] not in ICI.compiler_info:
-                ICI.compiler_info[details['compiler']]['includes'] = \
-                    ICI.get_compiler_includes(details['compiler'],
-                                              details['lang'],
+            # Independently of the actual compilation language in the
+            # compile command collect the iformation for C and C++.
+            if not ICI.compiler_info.get(compiler):
+
+                ICI.compiler_info[compiler] = defaultdict(dict)
+                # Collect for C
+                c_includes = \
+                    ICI.get_compiler_includes(compiler,
+                                              c_lang,
                                               details['analyzer_options'])
-                ICI.compiler_info[details['compiler']]['target'] = \
-                    ICI.get_compiler_target(details['compiler'])
-                ICI.compiler_info[details['compiler']]['default_standard'] = \
-                    ICI.get_compiler_standard(details['compiler'],
-                                              details['lang'])
-
-        details['compiler_includes'] = details['compiler_includes'] or \
-            ICI.compiler_info[details['compiler']]['includes']
-        details['compiler_standard'] = details['compiler_standard'] or \
-            ICI.compiler_info[details['compiler']]['default_standard']
-        details['target'] = details['target'] or \
-            ICI.compiler_info[details['compiler']]['target']
+                ICI.compiler_info[compiler][c_lang]['includes'] = c_includes
+                ICI.compiler_info[compiler][c_lang]['target'] = \
+                    ICI.get_compiler_target(compiler)
+                ICI.compiler_info[compiler][c_lang]['default_standard'] = \
+                    ICI.get_compiler_standard(compiler, c_lang)
+
+                # Collect for C++
+                ICI.compiler_info[compiler][cpp_lang]['includes'] = \
+                    ICI.get_compiler_includes(compiler, cpp_lang,
+                                              details['analyzer_options'])
+                ICI.compiler_info[compiler][cpp_lang]['target'] = \
+                    ICI.get_compiler_target(compiler)
+                ICI.compiler_info[compiler][cpp_lang]['default_standard'] = \
+                    ICI.get_compiler_standard(compiler, cpp_lang)
+
+        # Set the collected information for the build action.
+        details['compiler_includes'][c_lang] = \
```

Comment:
It seems quite easy to make an error here, perhaps using a function which takes three params would make it easier to see if there is an error.
Something like
`set('compiler_includes',c_lang,'includes')`